spec/lib/curtain_spec.rb

```diff
@@ -1,80 +1,71 @@
-require "spec_helper"
-
+require 'spec_helper'
 describe Rcurtain do
-
   before do
     Rcurtain.configure do |config|
       config.url = 'redis://:p4ssw0rd@10.0.1.1:6380/15'
     end
   end
-
   subject(:rcurtain) { Rcurtain.instance }
 
-  context "is opened" do
+  describe '#opened?' do
+    let(:user) { ['MPA-00000000000'] }
+    let(:users) { ['MPA-000000000000', 'MPA-111111111111'] }
 
-    context "when 100%" do
+    context 'when percentage is fully opened' do
       before do
         allow_any_instance_of(Redis).to receive(:get).and_return(100)
         allow_any_instance_of(Redis).to receive(:smembers).and_return(nil)
       end
 
-      it { expect(rcurtain.opened? "feature").to be true }
-    end
-
-    context "when user exists" do
-      before do
-        allow_any_instance_of(Redis).to receive(:get).and_return(0)
-        allow_any_instance_of(Redis).to receive(:smembers).and_return(['MPA-123'])
+      it 'should be true' do
+        expect(subject.opened?('feature')).to be true
       end
-
-      it { expect(rcurtain.opened?("feature", ['MPA-123'])).to be true }
     end
-  end
 
-  context "is closed" do
-    context "when 0%" do
-      before do
-        allow_any_instance_of(Redis).to receive(:get).and_return(0)
-        allow_any_instance_of(Redis).to receive(:smembers).and_return(nil)
+    context 'when percentage is fully closed' do
+      context 'and all users are enabled' do
+        before do
```

Comment:
**DRY**
Você pode deixar esse trecho do before dentro do contexto maior ('when percentage is fully closed'). Assim não precisa repetir ele em todos os testes d o mesmo contexto.